ts/components/wallet/__test__/OutcomeCodeMessageComponent.test.tsx

```diff
@@ -0,0 +1,101 @@
+import { View } from "native-base";
+import * as React from "react";
+import { NavigationParams } from "react-navigation";
+import configureMockStore from "redux-mock-store";
+import ROUTES from "../../../navigation/routes";
+import { applicationChangeState } from "../../../store/actions/application";
+import { appReducer } from "../../../store/reducers";
+import { GlobalState } from "../../../store/reducers/types";
+import { OutcomeCode } from "../../../types/outcomeCode";
+import { renderScreenFakeNavRedux } from "../../../utils/testWrapper";
+import OutcomeCodeMessageComponent from "../OutcomeCodeMessageComponent";
+
+const ASuccessComponent = () => <View testID="a-success-component"></View>;
+const ASuccessFooter = () => <View testID="a-success-footer"></View>;
+const onClose = jest.fn();
+
+describe("OutcomeCodeMessageComponent", () => {
+  beforeEach(() => jest.useFakeTimers());
+  it("Rendering OutcomeCodeMessageComponent, all the required components should be defined", () => {
+    const outcomeCode = { status: "success" } as OutcomeCode;
+    const { component } = renderComponent(
+      outcomeCode,
+      ASuccessComponent,
+      onClose
+    );
+
+    expect(component).not.toBeNull();
+  });
+  it("should render ASuccessFooter if outcomeCode status is equal to success and prop successFooter has been passed", () => {
+    const outcomeCode = { status: "success" } as OutcomeCode;
+    const { component } = renderComponent(
+      outcomeCode,
+      ASuccessComponent,
+      onClose,
+      ASuccessFooter
+    );
+
+    expect(component).not.toBeNull();
+  });
+  it("should render InfoScreenComponent if outcomeCode status is not equal to success and outcomeCode title is defined", () => {
+    const outcomeCode = {
+      status: "errorTryAgain",
+      title: {
+        "en-EN": "Test en",
+        "it-IT": "Test it"
+      }
+    } as OutcomeCode;
+    const { component } = renderComponent(
+      outcomeCode,
+      ASuccessComponent,
+      onClose
+    );
+
+    expect(component).not.toBeNull();
+    expect(component.queryByTestId("InfoScreenComponent")).not.toBeNull();
+  });
+  it("should render FooterWithButtons if outcomeCode status is equal to errorBlocking", () => {
+    const outcomeCode = {
+      status: "errorBlocking"
+    } as OutcomeCode;
+    const { component } = renderComponent(
+      outcomeCode,
+      ASuccessComponent,
+      onClose
+    );
+
+    expect(component).not.toBeNull();
+    expect(component.queryByTestId("FooterWithButtons")).not.toBeNull();
```

Comment:
great work 🚀 
Since we are here, what do you think to test also if title and description  are present and they are showing the right texts?

```suggestion
const outcomeCode = {
      status: "errorTryAgain",
      title: {
        "en-EN": "title en",
        "it-IT": "title it"
      },
      description: {
        "en-EN": "description en",
        "it-IT": "description it"
      }
    } as OutcomeCode;
    setLocale("it");
    const { component } = renderComponent(
      outcomeCode,
      ASuccessComponent,
      onClose
    );
    expect(component).not.toBeNull();
    const titleComponent = component.queryByText("title it");
    expect(titleComponent).toBeTruthy();
    const descriptionComponent = component.queryByText("description it");
    expect(descriptionComponent).toBeTruthy();
    expect(component.queryByTestId("InfoScreenComponent")).not.toBeNull();
```